pkg/splunk/test/controller.go

```diff
@@ -170,7 +170,10 @@ func (c MockClient) List(ctx context.Context, obj runtime.Object, opts ...client
 	})
 	listObj := c.ListObj
 	if listObj != nil {
-		copyMockObject(obj, listObj.(runtime.Object))
+		//if obj.GetObjectKind() == listObj.(runtime.Object).GetObjectKind() {
+		if reflect.TypeOf(obj).String() == reflect.TypeOf(listObj.(runtime.Object)).String() {
```

Comment:
Can we move this check to copyMockObject? That way any function which calls it with mis match of src and dst objects will be flagged with a log?